### ISSUE TITLE
duplicate module name

### ISSUE THREAD

**Issue by tjhance** [secure-foundations/dafny#50]
```dafny
abstract module A { }

abstract module A' refines A { }

module B(a: A) { }

module D(a: A) {
  import X = B(a)
}

module E(a: A') {
  import X = B(a)
}
```
gives error
```
Error: Modules 'B' and 'B' both have CompileName 'B_ON_a__Compile'.
```

**Comment by tjhance** [comment]
this is easy to workaround by renaming the module param, so it's not a high-priority. I think in general the names `B_ON_a__Compile` probably shouldn't take `a` from the user-supplied name.

It could be `B_ON_A__Compile` instead which both ensures uniqueness and prevents redundancy. (But we have to be careful if more than one parameter is a module-param refining the same module.)

**Comment by parno** [comment]
This appears to have also been fixed by 7302337
